genologics/lims.py

```diff
@@ -379,21 +380,24 @@ def _get_instances(self, klass, params=dict()):
             root = self.get(node.attrib['uri'], params=params)
         return result
 
-    def get_batch(self, instances):
-        "Get the content of a set of instances using the efficient batch call."
+    def get_batch(self, instances, force=False):
+        """Get the content of a set of instances using the efficient batch call."""
         if not instances:
             return []
-        klass = instances[0].__class__
+
         root = ElementTree.Element(nsmap('ri:links'))
+        klass = None
         result = []
         needs_request=False
         for instance in instances:
-            try:
-                result.append(self.cache[instance.uri])
-            except:
-                needs_request=True
+            if not klass:
```

Comment:
The instances returned by the batch request get a query string ?state=xxxx at the end of the URI, since the get_batch code actually takes the URI from the returned data. That is, the LIMS returns a different URI than the one we request, so when doing the batch call a second time, the original URIs won't be seen as cached.

I've made a commit to fix this, but I'm not sure if it's the right way. It uses the URIs of the instances provided as the argument to get_batch. On the plus side, get_batch now acts the same way as Entity.get(). However, it assumes that the batch call returns the data in exactly the same order as the request, and I can't find that written explicitly in the spec for get_batch. There's no way around this assumption if we don't add in a specific check for ?state.